tests/Doctrine/Tests/KeyValueStore/Storage/RiakStorageTest.php

```diff
@@ -67,182 +78,102 @@ public function testRequiresCompositePrimaryKeys()
-                $that->assertEquals('foobar', $key);
-                $that->assertEquals(['title' => 'Riak test'], $data);
-                return $objectMock;
-            }));
-
-        $this->storage->insert('riak-test', 'foobar', ['title' => 'Riak test']);
-    }
-
-    public function testUpdate()
-    {
-        $objectMock = $this->getMockBuilder('Riak\Object')
-            ->disableOriginalConstructor()
-            ->getMock();
-
-        $bucket = $this->getMockBuilder('Riak\Bucket')
-            ->disableOriginalConstructor()
-            ->getMock();
+        $data = [
+            'title' => 'Riak test',
+        ];
 
-        $this->riak->expects($this->once())
-            ->method('bucket')
-            ->will($this->returnValue($bucket));
+        $this->storage->insert('riak-test', 'foobar', $data);
 
-        $bucket->expects($this->once())
-             ->method('get')
-             ->will($this->returnValue($objectMock));
+        $namespace = new RiakNamespace('default', 'riak-test');
+        $location = new RiakLocation($namespace, 'foobar');
 
-        $that = $this;
-        $objectMock->expects($this->once())
-            ->method('setData')
-            ->will($this->returnCallback(function ($data) use ($that) {
-                $that->assertEquals(['title' => 'Riak cookbook'], $data);
-            }));
+        $fetch = FetchValue::builder($location)->build();
 
-        $objectMock->expects($this->once())
-            ->method('store');
+        $json = (string) $this->client
+            ->execute($fetch)
+            ->getValue()
+            ->getValue();
 
-        $this->storage->update('riak-test', 'foobar', ['title' => 'Riak cookbook']);
+        $this->assertSame($data, json_decode($json, true));
     }
 
-    public function testDelete()
+    /**
+     * @depends testInsert
+     */
+    public function testUpdate()
     {
-        $objectMock = $this->getMockBuilder('Riak\Object')
-            ->disableOriginalConstructor()
-            ->getMock();
+        $data = [
+            'title' => 'Riak update',
+        ];
 
-        $bucket = $this->getMockBuilder('Riak\Bucket')
-            ->disableOriginalConstructor()
-            ->getMock();
+        $this->storage->insert('riak-test', 'foobar', [
+            'title' => 'Riak insert',
+        ]);
 
-        $this->riak->expects($this->once())
-            ->method('bucket')
-            ->will($this->returnValue($bucket));
+        $this->storage->insert('riak-test', 'foobar', $data);
 
-        $bucket->expects($this->once())
-            ->method('get')
-            ->with('foobar')
-            ->will($this->returnValue($objectMock));
+        $namespace = new RiakNamespace('default', 'riak-test');
+        $location = new RiakLocation($namespace, 'foobar');
 
-        $objectMock->expects($this->once())
-            ->method('exists')
-            ->will($this->returnValue(true));
+        $fetch = FetchValue::builder($location)->build();
```

Comment:
The comment in `FetchValue` says

> Command used to fetch a value from Riak.

As far as I understand this will fetch one result value. Does this test cover having just one (updated) object stored in Riak?

Reply:
yes, it should check that the value related to the identifier `default.riak-test.foobar` is updated

Reply:
As far as I understand if an insert would be performed instead of an update, the test would still run without failure, because `FetchValue` just returns one entry.

Reply:
mmm I didn't see the double `insert` call, it should be ok now

Reply:
done in https://github.com/doctrine/KeyValueStore/pull/94/commits/13f7fb562423d1f9bdc944b3b4493fb230134f6e